src/backend/InvenTree/stock/models.py

```diff
@@ -458,44 +459,98 @@ def report_context(self):
+        Arguments:
+            serials: List of serial numbers to create
+            **kwargs: Additional keyword arguments to pass to the StockItem creation function
+
+        Returns:
+            QuerySet: The created StockItem objects
+
+        raises:
+            ValidationError: If any of the provided serial numbers are invalid
+
+        This method uses bulk_create to create multiple StockItem objects in a single query,
+        which is much more efficient than creating them one-by-one.
+
+        However, it does not perform any validation checks on the provided serial numbers,
+        and also does not generate any "stock tracking entries".
         """
-        serial = str(getattr(self, 'serial', '')).strip()
+        # Ensure the primary-key field is not provided
+        kwargs.pop('id', None)
+        kwargs.pop('pk', None)
+
+        part = kwargs.get('part')
+
+        if not part:
+            raise ValidationError({'part': _('Part must be specified')})
+
+        # Create a list of StockItem objects
+        items = []
+
+        # Provide some default field values
+        data = {**kwargs}
+
+        # Remove some extraneous keys which cause issues
+        for key in ['parent_id', 'part_id', 'build_id']:
+            data.pop(key, None)
+
+        data['parent'] = kwargs.pop('parent', None)
+        data['tree_id'] = kwargs.pop('tree_id', 0)
+        data['level'] = kwargs.pop('level', 0)
+        data['rght'] = kwargs.pop('rght', 0)
+        data['lft'] = kwargs.pop('lft', 0)
 
+        # Force single quantity for each item
+        data['quantity'] = 1
+
+        for serial in serials:
+            data['serial'] = serial
+            data['serial_int'] = StockItem.convert_serial_to_int(serial)
+
+            items.append(StockItem(**data))
+
+        # Create the StockItem objects in bulk
+        StockItem.objects.bulk_create(items)
+
+        # Return the newly created StockItem objects
+        return StockItem.objects.filter(part=part, serial__in=serials)
+
+    @staticmethod
+    def convert_serial_to_int(serial: str) -> int:
+        """Convert the provided serial number to an integer value.
+
+        This function hooks into the plugin system to allow for custom serial number conversion.
+        """
         from plugin.registry import registry
 
         # First, let any plugins convert this serial number to an integer value
         # If a non-null value is returned (by any plugin) we will use that
 
-        serial_int = None
-
         for plugin in registry.with_mixin('validation'):
             serial_int = plugin.convert_serial_to_int(serial)
 
             if serial_int is not None:
-                # Save the first returned result
-                # Ensure that it is clipped within a range allowed in the database schema
-                clip = 0x7FFFFFFF
-
-                serial_int = abs(serial_int)
-
-                serial_int = min(serial_int, clip)
```

Comment:
We had an issue with DoS attacks with too large serial numbers - that was fixed by this check; have you implemented an equivalent in the new code?

Reply:
Nice catch, I've re-implemented this now.